x/oracle/types/message_oracle.go

```diff
@@ -44,3 +63,104 @@ func (m *MsgRegisterOracle) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{oracleAddress}
 }
+
+var _ sdk.Msg = &MsgApproveOracleRegistration{}
+
+func NewMsgApproveOracleRegistration(approve *ApproveOracleRegistration, signature []byte) *MsgApproveOracleRegistration {
+	return &MsgApproveOracleRegistration{
+		ApproveOracleRegistration: approve,
+		Signature:                 signature,
+	}
+}
+
+func (m *MsgApproveOracleRegistration) Route() string {
+	return RouterKey
+}
+
+func (m *MsgApproveOracleRegistration) Type() string {
+	return "ApproveOracleRegistration"
+}
+
+func (m *MsgApproveOracleRegistration) ValidateBasic() error {
+	if err := m.ApproveOracleRegistration.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if m.Signature == nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "signature is empty")
+	}
+
+	return nil
+}
+
+func (m *MsgApproveOracleRegistration) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(m)
+	return sdk.MustSortJSON(bz)
+}
+
+func (m *MsgApproveOracleRegistration) GetSigners() []sdk.AccAddress {
+	oracleAddress, err := sdk.AccAddressFromBech32(m.ApproveOracleRegistration.ApprovedOracleAddress)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{oracleAddress}
+}
+
+func (m *ApproveOracleRegistration) ValidateBasic() error {
```

Comment:
Also, the `approver_oracle_address` and `target_oracle_address` can be checked if they are valid bech32 address